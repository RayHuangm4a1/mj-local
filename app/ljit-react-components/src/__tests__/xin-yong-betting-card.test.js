import React from 'react';
import { shallow, mount, } from 'enzyme';
import XinYongBettingCard from '../components/xin-yong-betting-card';

const {
	OrientationEnums,
	TypeEnums,
	XinYongPlaySlotTypeEnums,
} = XinYongBettingCard;

describe('XinYongBettingCard', () => {
	let playTemplates = [];
	let playsMap = {};
	let playSubcondition = {};

	beforeEach(() => {
		playTemplates = [
			{ id: 1, name: '大', },
			{ id: 2, name: '小', },
		];
		playsMap = {
			'1': { id: 1, name: '大', odds: 1954.00, },
			'2': { id: 2, name: '小', odds: 1954.00, },
		};
		playSubcondition = {
			name: '第一球',
			id: 1,
			bonus: 1.243,
			_id: '2131231',
		};
	});

	afterEach(() => {
		playTemplates = [];
		playsMap = {};
		playSubcondition = {};
	});

	it('should handle props', () => {
		const {
			playSubcondition,
			orientation,
			disabledText,
			isDisabled,
			defaultAmount,
			columnCount,
			playsMap,
			bettingsMap,
			playTemplates,
			xinYongPlaySlotType,
			onChange,
		} = XinYongBettingCard.defaultProps;

		expect(playSubcondition).toMatchObject({});
		expect(orientation).toEqual(OrientationEnums.VERTICAL);
		expect(disabledText).toEqual('');
		expect(isDisabled).toEqual(false);
		expect(defaultAmount).toEqual(0);
		expect(columnCount).toEqual(4);
		expect(playsMap).toMatchObject({});
		expect(bettingsMap).toMatchObject({});
		expect(playTemplates).toEqual([]);
		expect(xinYongPlaySlotType).toEqual(XinYongPlaySlotTypeEnums.LINE);
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			playTemplates: playTemplates,
			playsMap: playsMap,
			playSubcondition: playSubcondition,
			orientation: OrientationEnums.HORIZONTAL,
			disabledText: 'mock-text',
			isDisabled: false,
			defaultAmount: 10,
			columnCount: 4,
			onChange: () => {},
			className: 'mock-class',
			xinYongPlaySlotType: XinYongPlaySlotTypeEnums.LINE,
		};

		const wrapper = shallow(<XinYongBettingCard {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it ('should be selectable by class ljit-xin-yong-betting-card', () => {
		const wrapper = shallow(<XinYongBettingCard playTemplates={playTemplates} />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-card')).toEqual(true);
	});

	it ('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<XinYongBettingCard className={className} playTemplates={playTemplates} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it ('should be selectable by orientation is vertical', () => {
		const orientation = OrientationEnums.VERTICAL;
		const wrapper = shallow(<XinYongBettingCard orientation={orientation} playTemplates={playTemplates} />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-card--vertical')).toEqual(true);
	});

	it ('should be selectable by orientation is horizontal', () => {
		const orientation = OrientationEnums.HORIZONTAL;
		const wrapper = shallow(<XinYongBettingCard orientation={orientation} playTemplates={playTemplates} />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-card--horizontal')).toEqual(true);
	});

	it ('should mount in a full DOM', () => {
		const orientation = OrientationEnums.HORIZONTAL;
		const disabledText = 'mock-text';
		const className = 'mock-class';
		const isDisabled = false;
		const defaultAmount = 10;
		const columnCount = 4;
		const xinYongPlaySlotType = XinYongPlaySlotTypeEnums.LINE;
		const onChange = () => {};
		const wrapper = mount(
			<XinYongBettingCard
				playTemplates={playTemplates}
				playsMap={playsMap}
				playSubcondition={playSubcondition}
				orientation={orientation}
				disabledText={disabledText}
				className={className}
				isDisabled={isDisabled}
				defaultAmount={defaultAmount}
				columnCount={columnCount}
				xinYongPlaySlotType={xinYongPlaySlotType}
				onChange={onChange}
			/>
		);

		expect(wrapper.props().playTemplates).toEqual(playTemplates);
		expect(wrapper.props().playsMap).toEqual(playsMap);
		expect(wrapper.props().playSubcondition).toEqual(playSubcondition);
		expect(wrapper.props().orientation).toEqual(orientation);
		expect(wrapper.props().disabledText).toEqual(disabledText);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().isDisabled).toEqual(isDisabled);
		expect(wrapper.props().defaultAmount).toEqual(defaultAmount);
		expect(wrapper.props().columnCount).toEqual(columnCount);
		expect(wrapper.props().xinYongPlaySlotType).toEqual(xinYongPlaySlotType);
		expect(wrapper.props().onChange).toEqual(onChange);
	});
});

describe('XinYongBettingCard OrientationEnums', () => {
	it('should contain vertical property', () => {
		const typeName = 'vertical';
		const formatType = 'VERTICAL';

		expect(OrientationEnums[formatType]).toEqual(typeName);
	});

	it('should contain horizontal property', () => {
		const typeName = 'horizontal';
		const formatType = 'HORIZONTAL';

		expect(OrientationEnums[formatType]).toEqual(typeName);
	});
});

describe('XinYongBettingCard TypeEnums', () => {
	it('should contain CIRCLE property', () => {
		const typeName = 'circle';
		const formatType = 'CIRCLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains ROUND property', () => {
		const typeName = 'round';
		const formatType = 'ROUND';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contain RECTANGLE property', () => {
		const typeName = 'rectangle';
		const formatType = 'RECTANGLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contain ANIMAL property', () => {
		const typeName = 'animal';
		const formatType = 'ANIMAL';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});
});

describe('XinYongBettingCard XinYongPlaySlotTypeEnums', () => {
	it('should contain LINE property', () => {
		const typeName = 'line';
		const formatType = 'LINE';

		expect(XinYongPlaySlotTypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains SQUARE property', () => {
		const typeName = 'square';
		const formatType = 'SQUARE';

		expect(XinYongPlaySlotTypeEnums[formatType]).toEqual(typeName);
	});
});
