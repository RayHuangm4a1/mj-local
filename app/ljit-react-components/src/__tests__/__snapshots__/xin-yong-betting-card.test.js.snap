// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XinYongBettingCard should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XinYongBettingCard
    bettingsMap={Object {}}
    className="mock-class"
    columnCount={4}
    defaultAmount={10}
    disabledText="mock-text"
    isDisabled={false}
    onChange={[Function]}
    orientation="horizontal"
    playSubcondition={
      Object {
        "_id": "2131231",
        "bonus": 1.243,
        "id": 1,
        "name": "第一球",
      }
    }
    playTemplates={
      Array [
        Object {
          "id": 1,
          "name": "大",
        },
        Object {
          "id": 2,
          "name": "小",
        },
      ]
    }
    playsMap={
      Object {
        "1": Object {
          "id": 1,
          "name": "大",
          "odds": 1954,
        },
        "2": Object {
          "id": 2,
          "name": "小",
          "odds": 1954,
        },
      }
    }
    xinYongPlaySlotType="line"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-xin-yong-betting-card__title"
        >
          第一球
        </div>,
        <div
          className="ljit-xin-yong-betting-card__content"
        >
          <div
            className=""
          >
            <Row>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 1,
                      "name": "大",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 2,
                      "name": "小",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
            </Row>
          </div>
        </div>,
      ],
      "className": "ljit-xin-yong-betting-card ljit-xin-yong-betting-card--horizontal mock-class",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "第一球",
          "className": "ljit-xin-yong-betting-card__title",
        },
        "ref": null,
        "rendered": "第一球",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className=""
          >
            <Row>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 1,
                      "name": "大",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 2,
                      "name": "小",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
            </Row>
          </div>,
          "className": "ljit-xin-yong-betting-card__content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Row>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 1,
                      "name": "大",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
              <Col
                className="xin-yong-betting-card__col--4"
                span={6}
              >
                <XinYongPlaySlotLine
                  betting={Object {}}
                  defaultAmount={10}
                  disabledText="mock-text"
                  isDisabled={false}
                  onChange={[Function]}
                  play={
                    Object {
                      "id": 2,
                      "name": "小",
                      "odds": 1954,
                    }
                  }
                  playSlotType="circle"
                />
              </Col>
            </Row>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>,
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "xin-yong-betting-card-col-0",
                "nodeType": "class",
                "props": Object {
                  "children": <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />,
                  "className": "xin-yong-betting-card__col--4",
                  "span": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "xin-yong-play-slot-大1",
                  "nodeType": "class",
                  "props": Object {
                    "betting": Object {},
                    "defaultAmount": 10,
                    "disabledText": "mock-text",
                    "isDisabled": false,
                    "onChange": [Function],
                    "play": Object {
                      "id": 1,
                      "name": "大",
                      "odds": 1954,
                    },
                    "playSlotType": "circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "xin-yong-betting-card-col-1",
                "nodeType": "class",
                "props": Object {
                  "children": <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />,
                  "className": "xin-yong-betting-card__col--4",
                  "span": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "xin-yong-play-slot-小2",
                  "nodeType": "class",
                  "props": Object {
                    "betting": Object {},
                    "defaultAmount": 10,
                    "disabledText": "mock-text",
                    "isDisabled": false,
                    "onChange": [Function],
                    "play": Object {
                      "id": 2,
                      "name": "小",
                      "odds": 1954,
                    },
                    "playSlotType": "circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-xin-yong-betting-card__title"
          >
            第一球
          </div>,
          <div
            className="ljit-xin-yong-betting-card__content"
          >
            <div
              className=""
            >
              <Row>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
              </Row>
            </div>
          </div>,
        ],
        "className": "ljit-xin-yong-betting-card ljit-xin-yong-betting-card--horizontal mock-class",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "第一球",
            "className": "ljit-xin-yong-betting-card__title",
          },
          "ref": null,
          "rendered": "第一球",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className=""
            >
              <Row>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
              </Row>
            </div>,
            "className": "ljit-xin-yong-betting-card__content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Row>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
                <Col
                  className="xin-yong-betting-card__col--4"
                  span={6}
                >
                  <XinYongPlaySlotLine
                    betting={Object {}}
                    defaultAmount={10}
                    disabledText="mock-text"
                    isDisabled={false}
                    onChange={[Function]}
                    play={
                      Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      }
                    }
                    playSlotType="circle"
                  />
                </Col>
              </Row>,
              "className": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Col
                    className="xin-yong-betting-card__col--4"
                    span={6}
                  >
                    <XinYongPlaySlotLine
                      betting={Object {}}
                      defaultAmount={10}
                      disabledText="mock-text"
                      isDisabled={false}
                      onChange={[Function]}
                      play={
                        Object {
                          "id": 1,
                          "name": "大",
                          "odds": 1954,
                        }
                      }
                      playSlotType="circle"
                    />
                  </Col>,
                  <Col
                    className="xin-yong-betting-card__col--4"
                    span={6}
                  >
                    <XinYongPlaySlotLine
                      betting={Object {}}
                      defaultAmount={10}
                      disabledText="mock-text"
                      isDisabled={false}
                      onChange={[Function]}
                      play={
                        Object {
                          "id": 2,
                          "name": "小",
                          "odds": 1954,
                        }
                      }
                      playSlotType="circle"
                    />
                  </Col>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "xin-yong-betting-card-col-0",
                  "nodeType": "class",
                  "props": Object {
                    "children": <XinYongPlaySlotLine
                      betting={Object {}}
                      defaultAmount={10}
                      disabledText="mock-text"
                      isDisabled={false}
                      onChange={[Function]}
                      play={
                        Object {
                          "id": 1,
                          "name": "大",
                          "odds": 1954,
                        }
                      }
                      playSlotType="circle"
                    />,
                    "className": "xin-yong-betting-card__col--4",
                    "span": 6,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "xin-yong-play-slot-大1",
                    "nodeType": "class",
                    "props": Object {
                      "betting": Object {},
                      "defaultAmount": 10,
                      "disabledText": "mock-text",
                      "isDisabled": false,
                      "onChange": [Function],
                      "play": Object {
                        "id": 1,
                        "name": "大",
                        "odds": 1954,
                      },
                      "playSlotType": "circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "xin-yong-betting-card-col-1",
                  "nodeType": "class",
                  "props": Object {
                    "children": <XinYongPlaySlotLine
                      betting={Object {}}
                      defaultAmount={10}
                      disabledText="mock-text"
                      isDisabled={false}
                      onChange={[Function]}
                      play={
                        Object {
                          "id": 2,
                          "name": "小",
                          "odds": 1954,
                        }
                      }
                      playSlotType="circle"
                    />,
                    "className": "xin-yong-betting-card__col--4",
                    "span": 6,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "xin-yong-play-slot-小2",
                    "nodeType": "class",
                    "props": Object {
                      "betting": Object {},
                      "defaultAmount": 10,
                      "disabledText": "mock-text",
                      "isDisabled": false,
                      "onChange": [Function],
                      "play": Object {
                        "id": 2,
                        "name": "小",
                        "odds": 1954,
                      },
                      "playSlotType": "circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
